examples/knative/go/kn-go-tagging/README.md

```diff
@@ -0,0 +1,117 @@
+
+`ko` was created to help Knative developers build images and binaries without the need for Dockerfiles nor Makefiles.
+
+See more information in [this Knative blog](https://knative.dev/blog/articles/ko-fast-kubernetes-microservice-development-in-go/).
+
+Follow setup instructions at https://github.com/google/ko.
+
+You will also need to install [Go](https://golang.org/doc/install) for the ko tool to work.
+
+After installing ko, set the destination for images with an environment variable.
+
+```bash
+KO_DOCKER_REPO=my-dockerhub-user
+```
+
+Run the following command from the `kn-go-tagging` directory. It will build and push the image to your local Docker daemon. To have `ko` publish and use a container image registry, remove the `--local` flag.
+
+```bash
+ko publish --local .
+```
+
+Save the image name so you can use it in the [function.yaml]'s Service container image section (mentioned in Step 3).
+
+# Step 2 - Tests
+
+// TODO: put command here to run tests
+
+go tests
+
+# Step 3 - Deploy
+
+> **Note:** The following steps assume a working Knative environment using the
+> `default` Rabbit `broker`. The Knative `service` and `trigger` will be installed
+> in the `vmware-functions` Kubernetes namespace, assuming that the `broker` is
+> also available there.
+
+Push your container image to an accessible registry (if it wasn't already done in Step 1 - Build above).
+
+Create vSphere Credentials Secret
+
+Create a secret holding the username and password needed to access vCenter.
+
+```bash
+# create secret
+kubectl create secret generic vsphere-credentials \
+--from-literal=username=administrator@vsphere.local \
+--from-literal=password='ReplaceMe' \
+--namespace vmware-functions
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret tag-secret app=veba-ui
+```
+
+Edit the `function.yaml` file with the name of the container image from `Step 1`
+if you made any changes. If not, the default VMware container image will
+suffice.
+
+Get the name of the tag that will be attached to the VM by creating a tag within vSphere.
+Put the tag name in the `VALUE` field of `TAG_Name` for the Service manifest inside the `function.yaml` file.
+
+You can specify if the tag should be attached or detached to the VM that triggers the event by setting `TAG_ACTION` to `"attach"` or `"detach"` in the `function.yaml` Service manifest.
+
+These are all the configuration values that can be set in the Knative Service spec.template.spec.containers portion of the [Knative manifest](function.yaml):
+
+| Configuration       | Description                                                   | Example Values             | Required/Optional |
+| ------------------- | ------------------------------------------------------------- | -------------------------- | ----------------- |
+| image               | image name generated by ko                                    | ko.local/kn-go-tagging:0.1 | Required          |
```

Comment:
nit: might leave a not in the description that this can remain as is if no changes to the code were made